pkg/controller/certificate-shim/sync.go

```diff
@@ -508,6 +508,106 @@ func certNeedsUpdate(a, b *cmapi.Certificate) bool {
 		return true
 	}
 
+	if a.Spec.RevisionHistoryLimit != b.Spec.RevisionHistoryLimit {
+		return true
+	}
+
+	aAlgorithm, bAlgorithm := cmapi.RSAKeyAlgorithm, cmapi.RSAKeyAlgorithm
```

Comment:
Suggestion: do something like `var aAlgorithm, bAlgorithm campi.PrivateKeyAlgorithm` instead.

I understand that `RSA` is the default value so this check is logically correct, but is harder to read as the reader needs to follow that nil value in one cert is not different from RSA value in another etc.
Also this would follow the same approach as other checks in this function i.e [this](https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/certificate-shim/sync.go#L507) where we compare values as they are in the certs, not taking into account defaulting

Reply:
I actually tried this approach, but I found that this adds additional checks like `if aAlgorithm == "" && bAlgorithm != cmapi.RSAKeyAlgorithm` => `return true`.
Because we have to check the case when one private key does not have a value set (so it effectively has RSA as the default) and the other has RSA set implicitly.

Reply:
Fixed it